libimagcounter/src/counter.rs

```
@@ -140,21 +152,19 @@ impl<'a> Counter<'a> {
             })
     }
 
-    pub fn unit(&self) -> Result<CounterUnit> {
+    pub fn unit(&self) -> Option<CounterUnit> {
         self.fle.get_header().read("counter.unit")
-            .map_err(|e| CEK::StoreWriteError.into_error_with_cause(Box::new(e)))
-            .and_then(|u| match u {
-                Some(Value::String(s)) => Ok(CounterUnit(s)),
-                _ => Err(CEK::HeaderTypeError.into_error())
+            .ok()
+            .and_then(|s| match s {
+                Some(Value::String(s)) => Some(CounterUnit::new(s)),
+                _ => None,
```

Well, yes... this implementation makes sense as you're using it in `from_storeid()`, actually.

So I guess I'm fine with that, though I'm not fine with hiding the error. Returning an `Option::None` over an `Result:Err` is not fine, please revert the code to return an `Result` in case of error and adapt the rest of the codebase for it.